src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -1169,8 +1181,17 @@ void NetworkOPsImp::apply (std::unique_lock<std::mutex>& batchLock)
                     app_.overlay().foreach (send_if_not (
                         std::make_shared<Message> (tx, protocol::mtTRANSACTION),
                         peer_in_set(*toSkip)));
+                    e.transaction->setBroadcast();
                 }
             }
+
+            std::uint32_t accountSeq;
+            std::uint32_t availableSeq;
+            XRPAmount fee = app_.getTxQ().getTxRequiredFeeAndSeq(
+                *newOL, e.transaction->getSTransaction(),
+                accountSeq, availableSeq);
```

Comment:
I'd get rid of the return value combined with the out parameters and return a struct with all the value you want. You can then use the struct or use a structured binding here. Something like:
```
            auto [fee, accountSeq, availableSeq] =
                app_.getTxQ().getTxRequiredFeeAndSeq(
                    *newOL, e.transaction->getSTransaction());
```

Reply:
I'd like to package up all the the auxiliary ledger info and either set all of it or none of it. To support that, what do you think of setting the validated ledger index here. Something like:
```

            if (validatedLedgerIndex)
            {
                auto [fee, accountSeq, availableSeq] =
                    app_.getTxQ().getTxRequiredFeeAndSeq(
                        *newOL, e.transaction->getSTransaction());
                e.transaction->setCurrentLedgerState(
                    *validatedLedgerIndex, fee, accountSeq, availableSeq);
            }
```
Where `validatedLedgerIndex` is a `boost::optional` set outside the loop.

Reply:
Done.